reVX/plexos/rev_reeds_plexos.py

```diff
@@ -1,1460 +1,1134 @@
+        rev_join_on : str
+            Column name to join rev table.
+        reeds_join_on : str
+            Column name to join reeds table.
+        threshold : float
+            Maximum difference in coord matching.
+
+        Returns
+        -------
+        rev_sc : pd.DataFrame
+            Same as input.
+        reeds_build : pd.DataFrame
+            Same as input but without lat/lon columns if matched.
+        """
+
+        rev_coord_labels = get_coord_labels(rev_sc)
+        reeds_coord_labels = get_coord_labels(reeds_build)
+
+        if rev_coord_labels is not None and reeds_coord_labels is not None:
+            reeds_build = reeds_build.sort_values(reeds_join_on)
+            reeds_sc_gids = reeds_build[reeds_join_on].values
+            reeds_coords = reeds_build[reeds_coord_labels]
+
+            rev_mask = (rev_sc[rev_join_on].isin(reeds_sc_gids))
+            rev_sc = rev_sc.sort_values(rev_join_on)
+            rev_coords = rev_sc.loc[rev_mask, rev_coord_labels]
+
+            diff = reeds_coords.values - rev_coords.values
+            diff = diff.flatten()
+
+            if diff.max() > threshold:
+                emsg = ('reV SC and REEDS Buildout coordinates do not match. '
+                        'Max, mean coord distance: {}, {}'
+                        .format(diff.max(), diff.mean()))
+                logger.exception(emsg)
+                raise ValueError(emsg)
+
+            reeds_build = reeds_build.drop(labels=reeds_coord_labels, axis=1)
+
+        return rev_sc, reeds_build
+
+    def _make_forecast_map(self):
+        """Run ckdtree to map forecast pixels to generation pixels.
+
+        Returns
+        -------
+        fmap : np.ndarray | None
+            (n, 1) array of forecast meta data indices mapped to the generation
+            meta indices where n is the number of generation points. None if
+            no forecast filepath input.
+        """
+
+        fmap = None
+        if self._forecast_fpath is not None:
+            logger.info('Making KDTree from forecast data: {}'
+                        .format(self._forecast_fpath))
+            with Outputs(self._cf_fpath) as out:
+                meta_cf = out.meta
+            with Outputs(self._forecast_fpath) as out:
+                meta_fo = out.meta
+
+            clabels = get_coord_labels(meta_cf)
+            tree = cKDTree(meta_fo[clabels])
+            d, fmap = tree.query(meta_cf[clabels])
+            logger.info('Distance (min / mean / max) from generation pixels '
+                        'to forecast pixels is: {} / {} / {}'
+                        .format(d.min(), d.mean(), d.max()))
+
+        return fmap
+
+    def _make_node_map(self, k=1):
+        """Run ckdtree to map built rev SC points to plexos nodes.
+
+        Parameters
+        ----------
+        k : int
+            Number of neighbors to return.
+
+        Returns
+        -------
+        i : np.ndarray

```

Comment:
can we use something less ambiguous?

Reply:
This is the output var name of tree queries

Reply:
?

Reply:
https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.spatial.KDTree.query.html

Reply:
Still ambiguous, could we just do pos, or idx instead of i since most of our users are not Python wizes

Reply:
but im a wiz! lol okay, fixed, pushed.